dropshot/src/error.rs

```diff
@@ -118,6 +118,17 @@ pub struct HttpError {
     pub external_message: String,
     /// Error message recorded in the log for this error
     pub internal_message: String,
+    /// Headers which should be added to error responses generated from this
+    /// error.
+    ///
+    /// The [`http::HeaderMap`] is boxed to reduce the size of `HttpError`s
+    /// which don't contain a `HeaderMap` (the common case).  Unlike containers
+    /// such as `Vec` and `HashMap`, an empty [`http::HeaderMap`] contains a
+    /// fairly large number of fields, rather than being a pointer to a heap
+    /// allocation containing the actual data.  Thus, we store an
+    /// `Option<Box<HeaderMap>>` here, so that the empty case is just a null
+    /// pointer.
+    pub headers: Option<Box<http::HeaderMap>>,
```

Comment:
Boxing this resolves a Clippy lint about the error type being too large. I figured we shouldn't just disable that lint, since it would presumably also trigger on user code that returns an `HttpError`, and it seems sad to have an API that results in lints in consumers if they don't go out of their way to disable them.

I also considered making the field private, and requiring it be accessed via methods, since the `Option<Box<...>>` makes it a bit awkward. However, that would prevent consumers of the API from manually constructing `HttpError`s by initializing the other fields, which I've seen a lot of in Omicron, so I figured it was better to just make it public.